jetty-client/src/main/java/org/eclipse/jetty/client/GZIPContentDecoder.java

```diff
@@ -39,13 +47,47 @@ public GZIPContentDecoder(ByteBufferPool byteBufferPool, int bufferSize)
         super(byteBufferPool, bufferSize);
     }
 
+    @Override
+    public void beforeDecoding(HttpExchange exchange)
+    {
+        HttpFields.Mutable headers = exchange.getResponse().headers();
+        // Content-Length is not valid anymore while we are decoding.
+        headers.remove(HttpHeader.CONTENT_LENGTH);
+        // Content-Encoding should be removed/modified as the content will be decoded.
+        ListIterator<HttpField> fields = headers.listIterator();
+        while (fields.hasNext())
+        {
+            HttpField field = fields.next();
+            if (field.getHeader() == HttpHeader.CONTENT_ENCODING)
+            {
+                String value = Arrays.stream(field.getValues())
```

Comment:
Don't use a stream on critical path.    See server size, which has optimisation for the common gzip only `CONTENT_ENCODING`, otherwise it only accepts it as the last in the list of `CONTENT-ENCODING`s using a precompiled regex.     It is wrong to decode gzip if it is not the last encoding.

Reply:
It is `HttpReceiver` that has the logic to check for the last token of `Content-Encoding`, here we just want to remove it.

As for the usage of streams, I would otherwise have to allocate a `StringBuilder`, append to it, add some logic to add commas at the right places, so I'm not sure I'll save much?

Reply:
So if you have already checked that it is the last token, then don't tokenize it again here.   The code should be trivial:
```java
    int comma = field.getValue().lastIndexOf(',');
    if (comma == -1)
        fields.remove();
    else
        field.set(field.getValue().substring(0, comma);
```